Tests/ReadyCampaignDispatcherTests.swift

```diff
@@ -113,6 +170,20 @@ class ReadyCampaignDispatcherTests: QuickSpec {
                         dispatcher.dispatchAllIfNeeded()
                         expect(delegate.wasPingCalled).toEventually(beTrue())
                     }
+
+                    it("won't call shouldShowCampaignMessage delegate method for non-test campaigns") {
+                        let campaign = TestHelpers.generateCampaign(id: "test", title: "[ctx] title")
+                        dispatcher.addToQueue(campaign: campaign)
+                        dispatcher.dispatchAllIfNeeded()
+                        expect(delegate.wasShouldShowCalled).toAfterTimeout(beFalse())
```

Comment:
Why isn't `shouldShowCampaignMessage` called if it's a non-test campaign?

Reply:
In this context `when display permission is denied` it shouldn't. It would be pointless to ask host app about campaign that won't be displayed anyway

Reply:
Ah ok, I missed that context.